docs/form-guide.md

```diff
@@ -0,0 +1,104 @@
+# Form Guide
+
+## General content
+
+We use [`react-hook-form`](https://react-hook-form.com/) and [`zod`](https://github.com/colinhacks/zod#error-formatting) to create the form. Here are the basic steps to create a simple form
+
+### 1. Use `zod` to create form field schema
+
+You can set error message with i18n, you can read this file to learn about [i18n](i18n-guide.md)
+
+```js
+import { z as zod } from 'zod'
+
+export function ComponentForm() {
+  const t = useI18n() // Set errror messge with i18n.
+  const schema = zod.object({
+    name: zod.string(), // string
+    age: zod.number(t.needBeNumber).positive(t.needGreaterThanZero), // > 0
+    country: zod.string(t.countyNeedBeString).optional(), // string | undefiend
```

Comment:
```suggestion
  const t = useComponentI18N() // Set errror messge with i18n.
  const schema = zod.object({
    name: zod.string(), // string
    age: zod.number(t.needBeNumber()).positive(t.needGreaterThanZero()), // > 0
    country: zod.string(t.countyNeedBeString()).optional(), // string | undefiend
```

i reference this code snippet:

https://github.com/DimensionDev/Maskbook/blob/06147ebf244b3365368105ca210551971917fc23/packages/dashboard/src/pages/Settings/index.tsx#L33-L44